starter-core/src/test/groovy/io/micronaut/starter/feature/discovery/DiscoveryConsulSpec.groovy

```diff
@@ -31,6 +31,7 @@ class DiscoveryConsulSpec extends ApplicationContextSpec  implements CommandOutp
                 .render()
 
         then:
+        template.contains('implementation("io.micronaut:micronaut-discovery-core")')
```

Comment:
Use BuildVerifier API to test dependencies.

Reply:
@sdelamo The BuildTestVerifier API is in 3.9.x but not 4.0.x, which this PR has to target. Do you want to merge it up?